db/migrations/20200512130930_update_all_urns_function.sql

```diff
@@ -0,0 +1,83 @@
+-- +goose Up
+-- SQL in this section is executed when the migration is applied.
+
+DROP FUNCTION api.all_urns;
+
+CREATE FUNCTION api.all_urns(block_height bigint DEFAULT api.max_block(), max_results integer DEFAULT NULL::integer, result_offset integer DEFAULT 0) RETURNS SETOF api.urn_snapshot
+    LANGUAGE sql STABLE
+    AS $$
+    SELECT *
+    FROM ( SELECT DISTINCT ON (urn_identifier, ilk_identifier) urn_identifier, ilk_identifier,
+                              all_urns.block_height, ink, coalesce(art, 0), created, updated
+           FROM api.urn_snapshot
+           WHERE block_height <= all_urns.block_height
+           ORDER BY urn_identifier, ilk_identifier, updated DESC) AS latest_urns
```

Comment:
Can we just `ORDER BY updated` here (omitting `urn_identifier, ilk_identifier`)? Also, do we need `AS latest_urns`?

Reply:
Unfortunately we can't omit `urn_identifier` and `ilk_identifier` from ORDER BY because they are in the DISTINCT above. I'm not sure about latest_urns, I have to double-check.